tests/test_other.py

```diff
@@ -7771,7 +7775,7 @@ def test_EM_ASM_ES6(self):
   });
 }
 ''')
-    self.run_process([EMCC, 'src.cpp', '-O2'])
```

Comment:
should we keep the `-O2` (or add it above)? we run more optimizations with it, so removing it seems to remove coverage here.

Reply:
Actually it would be good to bump this to `-O3` even, to test more acorn optimizations.

Reply:
This was because JSDCE was eliminating the ES6 usage in the EM_ASM block.   I've updated it so that it gets preserved, and added variants of this test.